src/actions/fx-insert.js

```diff
@@ -82,31 +85,36 @@ export class FxInsert extends AbstractAction {
             // todo: eval 'at'
 
             /*
-            insert at position given by 'at' or use the last item in the targetSequence
-             */
+                        insert at position given by 'at' or use the last item in the targetSequence
+                         */
             // if (this.at) {
             if (this.hasAttribute('at')) {
                 // index = this.at;
                 // contextItem = targetSequence[this.at - 1];
 
-                index = evaluateXPath(this.getAttribute('at'), inscope, this.getOwnerForm());
+                // The 'at' attribute is an XPath that will resolv to a number
+                index = evaluateXPathToNumber(this.getAttribute('at'), inscope, this.getOwnerForm());
                 contextItem = targetSequence[index - 1];
             } else {
                 // this.at = targetSequence.length;
+
+                // TODO: when the target sequence is just the `data` element, this results in
+                // attempting to insert new nodes _next to_ that data element. Causing a DOM error:
+                // the data element resides directly under a document node.
+                // I guess we meant to fall back to the CHILDNODES of the target sequence here. But I am unsure where to add such a fallback.
```

Comment:
This is the TODO I mentioned

Reply:
Problem is with inhomogenius sequences consisting of nodes but also atomics -these are allowed by XForms 2 and that brought me to use evaluateXPath must i already saw the trouble with that. Not sure right now how to best deal with it. 

I need to look it up again but i guess in the above case we are expected to insert that element into data then. I've not tested more complicated cases yet.

Regarding data element - following the strict logic a `<fx-insert ref="task">` should have no effect if there is no `origin ` given (as we do not have a 'template' to insert). Looking at it that way would mean the test has the wrong title. Should actually be something like 'ignores insert when there's no origin and target sequence is empty'.

To be honest i haven't digged through all that spec text about insert. that's horrible stuff ;) but for the above case i'm pretty sure as the mechanics are always 'use the last element and clone it as new item to insert OR use whatever is referred to by origin'.

Does that sound reasonable?